In [1]:
import warnings
warnings.filterwarnings("ignore")
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import env
import wrangle_zillow
import os

In [2]:
zillow2017_df=wrangle_zillow.wrangle_zillow()

In [3]:
zillow2017_df.isnull().sum()

bedroomcnt                        11
bathroomcnt                       11
calculatedfinishedsquarefeet    8484
taxvaluedollarcnt                493
yearbuilt                       9337
taxamount                       4442
fips                               0
dtype: int64

In [4]:
zillow2017_df.describe().T

count           mean            std  \
bedroomcnt                    2152852.0       3.287196       0.954754   
bathroomcnt                   2152852.0       2.230688       0.999280   
calculatedfinishedsquarefeet  2144379.0    1862.855178    1222.125124   
taxvaluedollarcnt             2152370.0  461896.237963  699676.049600   
yearbuilt                     2143526.0    1960.949681      22.162196   
taxamount                     2148421.0    5634.865978    8178.910249   
fips                          2152863.0    6048.377335      20.433292   

                                  min        25%        50%        75%  \
bedroomcnt                       0.00       3.00       3.00       4.00   
bathroomcnt                      0.00       2.00       2.00       3.00   
calculatedfinishedsquarefeet     1.00    1257.00    1623.00    2208.00   
taxvaluedollarcnt                1.00  188170.25  327671.00  534527.00   
yearbuilt                     1801.00    1949.00    1958.00    1976.00   
taxamount                        1.85    2534.98    4108.95    6414.32   
fips                          6037.00    6037.00    6037.00    6059.00   

                                      max  
bedroomcnt                          25.00  
bathroomcnt                         32.00  
calculatedfinishedsquarefeet    952576.00  
taxvaluedollarcnt             98428909.00  
yearbuilt                         2016.00  
taxamount                      1337755.86  
fips                              6111.00

In [5]:
zillow2017_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2152863 entries, 0 to 2152862
Data columns (total 7 columns):
 #   Column                        Dtype  
---  ------                        -----  
 0   bedroomcnt                    float64
 1   bathroomcnt                   float64
 2   calculatedfinishedsquarefeet  float64
 3   taxvaluedollarcnt             float64
 4   yearbuilt                     float64
 5   taxamount                     float64
 6   fips                          float64
dtypes: float64(7)
memory usage: 115.0 MB


# Key takeaway
There are 2152863 data returned, there are multiple null values within each columns except fips.

In [6]:
def remove_outliers(zillow2017_df, k, col_list):
    for col in col_list:
        q1, q3 = zillow2017_df[col].quantile([.25, .75]) 
        iqr = q3 - q1  
        upper_bound = q3 + k * iqr  
        lower_bound = q1 - k * iqr  
        zillow2017_df = zillow2017_df[(zillow2017_df[col] > lower_bound) & (zillow2017_df[col] < upper_bound)]
    return zillow2017_df

In [7]:
# use previous defined function to remove the outliers
zillow2017_df = remove_outliers(zillow2017_df, 1.5, ['bedroomcnt', 'bathroomcnt', 'calculatedfinishedsquarefeet', 'taxvaluedollarcnt', 'taxamount'])
zillow2017_df

bedroomcnt  bathroomcnt  calculatedfinishedsquarefeet  \
6               3.0          4.0                        1620.0   
7               3.0          2.0                        2077.0   
18              3.0          1.0                        1244.0   
19              3.0          2.0                        1300.0   
20              3.0          2.0                        1222.0   
...             ...          ...                           ...   
2152852         3.0          2.0                        1917.0   
2152853         4.0          2.0                        1987.0   
2152855         3.0          2.5                        1809.0   
2152861         3.0          2.5                        1974.0   
2152862         4.0          4.0                        2110.0   

         taxvaluedollarcnt  yearbuilt  taxamount    fips  
6                 847770.0     2011.0   10244.94  6037.0  
7                 646760.0     1926.0    7924.68  6037.0  
18                169471.0     1950.0    2532.88  6037.0  
19                233266.0     1950.0    3110.99  6037.0  
20                290492.0     1951.0    3870.25  6037.0  
...                    ...        ...        ...     ...  
2152852           408680.0     1946.0    4341.32  6111.0  
2152853           259913.0     1955.0    3175.66  6059.0  
2152855           405547.0     2012.0    4181.10  6059.0  
2152861           424353.0     2015.0    5302.70  6059.0  
2152862           554009.0     2014.0    6761.20  6037.0  

[1855285 rows x 7 columns]

In [8]:
# rename the columns to simplify the name
zillow2017_df.rename(columns={'bedroomcnt':'bedroom','bathroomcnt':'bathroom','calculatedfinishedsquarefeet':'sqtft','taxvaluedollarcnt':'taxvalue'},inplace=True)

In [9]:
zillow2017_df.isnull().sum()

bedroom        0
bathroom       0
sqtft          0
taxvalue       0
yearbuilt    874
taxamount      0
fips           0
dtype: int64

# Key Takeaway
After remove outlier, the dataset has 1855285 rows remained and 874 null values